src/sagemaker/fw_utils.py

```diff
@@ -418,7 +421,13 @@ def tar_and_upload_dir(
     script_name = script if directory else os.path.basename(script)
     dependencies = dependencies or []
     key = "%s/sourcedir.tar.gz" % s3_key_prefix
-    tmp = tempfile.mkdtemp()
+    if local_download_dir is not None and not (
+        os.path.exists(local_download_dir) and os.path.isdir(local_download_dir)
+    ):
+        raise ValueError(
+            f"Inputted directory for storing newly generated temporary directory does not exist: '{local_download_dir}'"
+        )
```

Comment:
how about attempting to create the target dir with `os.makedirs(dir, exists_ok=True)` instead?

Reply:
I'd rather not. Making directories on behalf of customers seems risky.